libsignal-protocol/src/context.rs

```diff
@@ -309,6 +308,14 @@ impl Default for Context {
     }
 }
 
+// Assert that Context does not implement [`Send`] and [`Sync`] to guarantee
+// that each [`Context`] instance is only used within a single thread. This allows
+// locking to be omitted (see lock_function below).
+//
+// See https://github.com/Michael-F-Bryan/libsignal-protocol-rs/issues/49
+// for details.
+static_assertions::assert_not_impl_all!(Context: Send, Sync);
```

Comment:
Could this line be moved directly above `lock_function()`? That way our assertion that `Context: !Send + !Sync` is right next to the comment that says locking isn't required as long as `Context` doesn't implement `Send` and `Sync`.

Reply:
Agree. This looks better.

Fixed in eddb1cd1d